context/app/static/js/components/detailPage/AnalysisDetails/AnalysisDetails.tsx

```diff
@@ -1,33 +1,233 @@
+  const pipelineName = name ?? origin.split('/').pop();
+
+  const githubUrl = buildGithubURL(pipeline);
+  const cwlUrl = `https://view.commonwl.org/workflows/${githubUrl.replace(/^http(s?):\/\//i, '')}`;
+
+  const onExpand = useCallback(
+    (isExpanded: boolean) => {
+      if (isExpanded) {
+        setRowIsExpanded(true);
+      }
+    },
+    [setRowIsExpanded],
+  );
+
+  return (
+    <ExpandableRow
+      numCells={7}
+      expandedContent={input_parameters ? <WorkflowParameters input_parameters={input_parameters} /> : null}
+      reverse
+      disabled={!input_parameters?.length}
+      onExpand={onExpand}
+    >
+      <ExpandableRowCell>{i + 1}</ExpandableRowCell>
+      <ExpandableRowCell>
+        {pipelineName} {Boolean(version) && `(${version})`}
+      </ExpandableRowCell>
+      <ExpandableRowCell>
+        <OutboundIconLink href={origin}>{origin}</OutboundIconLink>
+      </ExpandableRowCell>
+      <ExpandableRowCell>
+        <Box>
+          <OutboundIconLink href={githubUrl}>{hash}</OutboundIconLink>
+        </Box>
+      </ExpandableRowCell>
+      <ExpandableRowCell>
+        {documentation_url ? <OutboundLink href={documentation_url}>{documentation_url}</OutboundLink> : '-'}
+      </ExpandableRowCell>
+      <ExpandableRowCell>
+        {cwlUrl.endsWith('cwl') ? (
+          <Box>
+            <OutboundLinkButton href={cwlUrl} component="a" variant="outlined" color="info">
+              Open CWL Viewer
+            </OutboundLinkButton>
+          </Box>
+        ) : (
+          '-'
+        )}
+      </ExpandableRowCell>
+    </ExpandableRow>
+  );
+}
+
+const cols = ['Step', 'Tool', 'Origin Link', 'Git Commit', 'Documentation Link', null];
+
+function WorkflowSteps({
+  steps,
+  setRowIsExpanded,
+}: {
+  steps: Pipeline[];
+  setRowIsExpanded: Dispatch<SetStateAction<boolean>>;
+}) {
+  return (
+    <StyledTableContainer component={Paper} maxHeight={550}>
+      <Table stickyHeader>
+        <TableHead>
+          <TableRow>
+            <TableCell sx={{ backgroundColor: 'background.paper' }}>&nbsp; {/* Expansion cell column */} </TableCell>
+            {cols.map((col) => (
+              <TableCell sx={{ backgroundColor: 'background.paper' }} key={col}>
+                {col}
+              </TableCell>
+            ))}
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {steps.map((pipeline, i) => (
+            <WorkflowStep
+              pipeline={pipeline}
+              i={i}
+              setRowIsExpanded={setRowIsExpanded}
+              key={pipeline.name + pipeline.origin + pipeline.hash + pipeline.version}
+            />
+          ))}
+        </TableBody>
+      </Table>
+    </StyledTableContainer>
+  );
+}
+
+function dedupeSteps(steps: Pipeline[]) {
+  const stepMap = steps.reduce<Record<string, Pipeline>>((acc, curr) => {
+    const { name, hash, origin, version } = curr;
+
+    const key = [name, hash, origin, version].join();
```

Comment:
Small idea for consistency/future clarity, can probably be left as is. We'll also need to hoist the createPipelineKey function above the `WorkflowSteps` component to avoid the lint error, but I wanted to keep things readable for the comment:

```suggestion
              key={createPipelineKey(pipeline)}
            />
          ))}
        </TableBody>
      </Table>
    </StyledTableContainer>
  );
}

function createPipelineKey({ name, hash, origin, version }: Pipeline) {
  return [name, hash, origin, version].join();
}

function dedupeSteps(steps: Pipeline[]) {
  const stepMap = steps.reduce<Record<string, Pipeline>>((acc, curr) => { 
    const key = createPipelineKey(curr)
```